### ISSUE TITLE
[Feature Request]: Install PHP CLI binaries in versioned folders to support PATH manipulation for isolated sites

### ISSUE THREAD

**Issue by leewillis77** [beyondcode/herd-community#103]
### Feature Description

On my development box I have a number of different projects that have different PHP requirements. Herd's isolation functionality neatly handles making sure that sites are served with the correct version, but it's difficult to ensure that the right PHP CLI binary is invoked for a project. 

Under valet/homebrew I handled this with an autoenv script which prepended the version specific PHP folder to the PATH. Since homebrew installs the binaries in versioned folders, this worked nicely. E.g. the following autoenv script ensures that when I'm in that project folder, my PATH will cause php 8.0 binary to be used for CLI commands. 

```sh
php_version=`ls -1rt /opt/homebrew/Cellar/php@8.0/ | tail -1`
autostash PATH=/opt/homebrew/Cellar/php@8.0/"$php_version"bin:$PATH
```

Most important about this solution is that it's local to that shell - ie I can have two terminals open in two different projects, and each will invoke the correct PHP binary. 

I'm aware of `herd use`, but that doesn't really help since that's global. 

It would be nice if herd created versioned folders within `/Users/lee/Library/Application\ Support/Herd/bin/` so that this use-case could be supported. The naming isn't ideal, but here's an example structure that would work:

```sh
$ ls -lR /Users/lee/Library/Application\ Support/Herd/bin/versions
total 0
drwxr-xr-x  3 lee  staff    96B  7 Aug 10:44 php74/
drwxr-xr-x  3 lee  staff    96B  7 Aug 10:44 php80/
drwxr-xr-x  3 lee  staff    96B  7 Aug 10:44 php81/
drwxr-xr-x  3 lee  staff    96B  7 Aug 10:44 php82/

/Users/lee/Library/Application Support/Herd/bin/versions/php74:
total 0
lrwxr-xr-x  1 lee  staff    53B  7 Aug 10:44 php@ -> /Users/lee/Library/Application Support/Herd/bin/php74

/Users/lee/Library/Application Support/Herd/bin/versions/php80:
total 0
lrwxr-xr-x  1 lee  staff    53B  7 Aug 10:44 php@ -> /Users/lee/Library/Application Support/Herd/bin/php80

/Users/lee/Library/Application Support/Herd/bin/versions/php81:
total 0
lrwxr-xr-x  1 lee  staff    53B  7 Aug 10:44 php@ -> /Users/lee/Library/Application Support/Herd/bin/php81

/Users/lee/Library/Application Support/Herd/bin/versions/php82:
total 0
lrwxr-xr-x  1 lee  staff    53B  7 Aug 10:44 php@ -> /Users/lee/Library/Application Support/Herd/bin/php82
```

With a structure like this, each project folder can have its PATH linked to a version-specific folder, and hence invoke the correct PHP version. 

### Is this feature valuable for other users as well and why?

Supports anyone working on multiple projects each with differing PHP requirements.

**Comment by thomas-alrek** [comment]
I'm in the exact same situation. I maintain projects raning from 7.4 to 8.2. I really like using Herd, but it get's a bit tiring having to manually switch the default PHP version when using the CLI for updating dependencies etc. It would be amazing if the Herd PHP binary was able to automatically switch to the correct PHP version per site, respecting the version that's selected in the UI for the CLI as well.

**Comment by mpociot** [comment]
You don't have to switch the default php version. If the site is isolated you can use "herd php" to redirect calls to the correct php version. 
Likewise you may use "herd composer". 
If you just want to access the other binaries without linking your site, they are globally available as: php74, php80, php81, php82, etc.

**Comment by leewillis77** [comment]
@thomas-alrek I think having herd handle the switching is a bit beyond its scope (after all there are a myriad of terminals & shells out there). It's certainly beyond the scope of what I was asking for here.

**Comment by thomas-alrek** [comment]
Yeah probably. But since Herd already provide a PHP proxy binary that redirects to the correct version based on your preferences, I feel like it could be in scope. It shouldn't be too hard to implement in theory. Would be very convenient

**Comment by leewillis77** [comment]
@thomas-alrek It should maybe a separate issue then, because it's different for what this is asking for.
